snf-cyclades-app/synnefo/api/util.py

```diff
@@ -190,6 +191,13 @@ def get_image(image_id, user_id):
             raise faults.ItemNotFound("Image '%s' not found" % image_id)
 
 
+def get_keypair(keypair_name):
+    try:
+        return PublicKeyPair.objects.get(name=keypair_name)
```

Comment:
What if two users have a key with the same name? You need to also check for the user id.